### ISSUE TITLE
[gatsby-source-contentful] downloadLocal not working

### ISSUE THREAD

**Issue by ascabknight** [gatsbyjs/gatsby#18483]
## Description

In our company, we have built a web site using Gatsby and Contentful. Our CI has been experiencing perfromance issues with the build time. Running a build in Production is taking around 1.5 - 2 hours to complete (yes, it's a large website). Debugging the build process, I noticed that the Contentful assets were being downloaded despite the _downloadLocal_ option was being set as "false". 

Taking a look at the _gatsby-source-contentful_ source code, I noticed that the _downloadLocal_ option is not being validated correctly, the _get_ method from the _publishConfig_ object is returning a string value thus the following condition will always be true. This is even ignoring the default value of the downloadLocal option.

_gatsby-node.js (Line 225)_

```
//This will always be true since a string is being returned not a boolean.
if (pluginConfig.get('downloadLocal')) {
    await downloadContentfulAssets({
      actions,
      createNodeId,
      store,
      cache,
      getNodes,
      reporter,
    })
}
```

### Steps to reproduce

1. Setup the _gatsby-source-contentful_ and _gatsby-source-filesystem_ folder plugins in your website.
2. Set the downloadLocal option as false.
3. Run a _gatsby build_
4. Take a look at the .cache folder and you will notice that assets are being downloaded into the gatsby-source-filesystem folder.
5. Open the gatsby-node.js file from the gatsby-source-contentful plugin and run the following test. You will notice the results immediately:

**Test**

```
const downloadLocal = pluginConfig.get(`downloadLocal`)

//This will return true or false depending on your settings or the default value
console.error(`****** gatsby-source-contentful/downloadLocal: ${downloadLocal}. ******`)

//This returns: string.
console.log(`typeof: ${typeof downloadLocal }`)

// Double checking the downloadLocal flag, you can improve this implementation but for testing it works.
  if (downloadLocal && downloadLocal === true) {
    await downloadContentfulAssets({
      actions,
      createNodeId,
      store,
      cache,
      getNodes,
      reporter
    });
  }
```

### Expected result

- Setting downloadLocal as true should download assets locally.
- Setting downloadLocal as false should NOT download assets locally.

### Actual result

Setting downloadLocal as false is still downloading contentful assets in the local folder impacting our build times.

### Environment

Run `gatsby info --clipboard` in your project directory and paste the output here.
```
  System:
    OS: macOS 10.14.5
    CPU: (12) x64 Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz
    Shell: 5.3 - /bin/zsh
  Binaries:
    Node: 10.16.3 - ~/.nvm/versions/node/v10.16.3/bin/node
    Yarn: 1.17.3 - /usr/local/bin/yarn
    npm: 6.9.0 - ~/.nvm/versions/node/v10.16.3/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 77.0.3865.90
    Firefox: 68.0
    Safari: 12.1.1
  npmPackages:
    gatsby: ^2.15.30 => 2.15.35 
    gatsby-image: ^2.2.7 => 2.2.7 
    gatsby-plugin-eslint: ^2.0.4 => 2.0.5 
    gatsby-plugin-google-tagmanager: ^2.1.3 => 2.1.4 
    gatsby-plugin-i18next: ^1.1.7 => 1.1.7 
    gatsby-plugin-react-helmet: ^3.1.1 => 3.1.2 
    gatsby-plugin-react-svg: ^2.1.1 => 2.1.1 
    gatsby-plugin-resolve-src: ^2.0.0 => 2.0.0 
    gatsby-plugin-robots-txt: ^1.5.0 => 1.5.0 
    gatsby-plugin-sharp: ^2.2.5 => 2.2.9 
    gatsby-plugin-styled-components: ^3.1.1 => 3.1.2 
    gatsby-source-contentful: ^2.1.11 => 2.1.15 
    gatsby-source-filesystem: ^2.1.4 => 2.1.6 
    gatsby-transformer-remark: ^2.6.3 => 2.6.9 
    gatsby-transformer-sharp: ^2.2.2 => 2.2.4 

```

**Comment by samuelgoldenbaum** [comment]
To add: I get the following logged when setting `downloadLocal: true` and running `gatsby build`

```
 ERROR 

(node:60875) [DEP0066] DeprecationWarning: OutgoingMessage.prototype._headers is deprecated
```

**Comment by AnalogMemory** [comment]
I ran into this issue today with node `12.13.0`, temporarliy solved by downgrading to node `10.17.0`

It's an issue with a dependency in `gatsby-source-filesystem` see [#18857](https://github.com/gatsbyjs/gatsby/pull/18857)
Looks like it'll be fixed in a few months
